compiler/rustc_codegen_ssa/src/mir/intrinsic.rs

```diff
@@ -369,6 +369,16 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 }
             }
 
+            sym::const_allocate => {
+                // just returns a null pointer at runtime.
+                bx.const_null(bx.type_i8p())
```

Comment:
This does not seem right -- it should abort or panic or something like that.

Reply:
We could also add an opt to optimized_mir to inline const_eval_select calls